kedro/extras/datasets/api/api_dataset.py

```diff
@@ -107,12 +90,13 @@ def _execute_request(self) -> requests.Response:
         return response
 
     def _load(self) -> requests.Response:
-        return self._execute_request()
+        with sessions.Session() as session:
+            return self._execute_request(session)
```

Comment:
This is totally fine, I'm just curious why you chose to explicitly create the `session` given this seems to be what's done under the hood already by `requests`. I'm not very familiar with making requests with or without a sessions so would like to learn! Do you think it's just nice to be more explicit?

Reply:
I did this because subclasses (the API Paginator) can then use this session to do more requests without overhead.

Reply:
Sounds good, thank you!